autoendpoint/src/routers/fcm/client.rs

```diff
@@ -34,8 +33,7 @@ impl FcmClient {
     ) -> std::io::Result<Self> {
         // `map`ping off of `serde_json::from_str` gets hairy and weird, requiring
         // async blocks and a number of other specialty items. Doing a very stupid
-        // json detection does not. GCM keys are base64 values, so  `{` will never
-        // appear in a GCM key. FCM keys are serialized JSON constructs.
+        // json detection does not. FCM keys are serialized JSON constructs.
```

Comment:
What does the accompanying change to GCM settings look like?

Obviously an `is_gcm` will be added to GCM project ids. Per the previous comment I expect the current GCM "credential" setting to be set to a value not including "{". It's definitely not a path (it was the Authorization header's `key=` value), so it would produce a `None` authenticator.

A test can confirm this, I made the `unknown_fcm_error` test panic with this change:

```diff
diff --git a/autoendpoint/src/routers/fcm/client.rs b/autoendpoint/src/routers/fcm/client.rs
index bfbb8ba..1d5a44f 100644
--- a/autoendpoint/src/routers/fcm/client.rs
+++ b/autoendpoint/src/routers/fcm/client.rs
@@ -367,8 +367,8 @@ pub mod tests {
     async fn unknown_fcm_error() {
         let client = make_client(FcmServerCredential {
             project_id: PROJECT_ID.to_owned(),
-            is_gcm: Some(false),
-            server_access_token: make_service_key(),
+            is_gcm: Some(true),
+            server_access_token: "AIzaSyB0ecSrqnEDXQ7yjLXqVc0CUGOeSlq9BsM".to_owned(),
         })
         .await;
         let _token_mock = mock_token_endpoint();
```

Reply:
Oh it's in your original comment, sorry I missed it.

Reply:
We don't want to use `server_access_token`, since that is no longer valid. Instead we want to use the OAuth credential set, similar to how we currently do FCM credentials, with those credentials tied to the main FCM "boxwood" authset, which are a JSON structure.

Changing ` is_gcm: Some(true), server_access_token: make_service_key() ` does not panic.

I've modified the initial git commit message to match the edited commit text above.